packages/roosterjs-editor-plugins/lib/plugins/ContentEdit/features/listFeatures.ts

```diff
@@ -92,6 +94,30 @@ const OutdentWhenBackOn1stEmptyLine: BuildInEditFeature<PluginKeyboardEvent> = {
     handleEvent: toggleListAndPreventDefault,
 };
 
+/**
+ * MaintainListChainWhenDelete edit feature, provides the ability to indent the list if user press
+ * DELETE before the first item of a list
+ */
+const MaintainListChainWhenDelete: BuildInEditFeature<PluginKeyboardEvent> = {
+    keys: [Keys.DELETE],
+    shouldHandleEvent: (event, editor) => {
+        const li = editor.getElementAtCursor('LI', null /*startFrom*/, event);
+        if (li) {
+            return false;
+        }
+        const nextSibiling = getCacheNextSibiling(event, editor);
+        const nextSibilingIsLI = findClosestElementAncestor(nextSibiling, null, 'li');
+        const isAtEndAndBeforeLI = nextSibilingIsLI
+            ? Position.getEnd(editor.getSelectionRange()).isAtEnd
+            : false;
```

Comment:
I prefer do this check first since it is cheaper than the traverser thing.

So first, check is in LI, second, check if is at end, third, check if next block is LI.

We already do cheapest check first, and let it fail fast.